### DISCUSSION TITLE
How to download map for current view/location

### DISCUSSION THREAD

**Discussion by thompehw** [osmandapp/OsmAnd#12894] [Q&A] [answered]
Android OsmAnd+ 4.0.8

In earlier versions, when I zoomed-in on a part of the world for which I had no local map there was a banner (sometimes annoying) suggesting  which map to download.

Now when I want to investigate a far-away place (for an OSM mapping issue) I can't find an easy way to discover which map to download. Even if I drop a pin I haven't been able to find an option for it. 

If I say I want to navigate to the place, then I do find an option to download all the maps between where I am and the far-away place.

How do I easily download a map for the place I'm looking at? Sorry if I'm missing something obvious - I hope I am!

**Comment by dmpr0** [comment]
Hi!
Download dialog on the top part of the screen should appear from zoom level 9, when base map data is no longer available. You can check if this dialog doesn't disabled in Menu – Settings – OsmAnd settings – Dialogs and notifications.
Long tap on the empty area should open content menu with the download button, if it doesn't happen, try to open Download Maps screen and tap on "update" icon, this need download file with available maps and regios.

**Reply by tommycrock** [reply]
I see the same as @thompehw 
![Screenshot_20210930-132007_OsmAnd+](https://user-images.githubusercontent.com/5749884/135454376-4ad3824d-021c-4542-b979-c04342f0556e.jpg)
![Screenshot_20210930-132028_OsmAnd+](https://user-images.githubusercontent.com/5749884/135454379-5da7d5f6-d91f-46a5-a357-81d81fbe1f48.jpg)
![Screenshot_20210930-132644_OsmAnd+](https://user-images.githubusercontent.com/5749884/135454690-a6be02ff-cfb1-4fc2-8ae2-184069f5de64.jpg)

**Reply by thompehw** [reply]
Thanks @dmpr0 ! Following your advice I still don't see a dialog at the top of the map screen (as for  @tommycrock ) but I can find buttons after a long-press. They don't appear if I stay in landscape mode, and they don't work! Here are some pictures:

Check that it's enabled (it was):
![image002](https://user-images.githubusercontent.com/39751382/135737075-221d6e8e-2e5f-4055-9b44-8ea24f248b39.jpg)
Move to an area for which I have no map (Paris):
![image003](https://user-images.githubusercontent.com/39751382/135737184-6755d7cf-e3f1-4309-b920-b5ddf5b40cdd.jpg)
Zoom in to level 10 - nothing appears on the map:
![image004](https://user-images.githubusercontent.com/39751382/135737206-351aebe1-b876-4625-bbdc-891f88b1f7d8.jpg)
Long press:
![image005](https://user-images.githubusercontent.com/39751382/135737213-db97d199-0ec1-4dae-a7a8-2c7ba913e116.jpg)
Choose actions- there are no map download options:
![image006](https://user-images.githubusercontent.com/39751382/135737229-5fe01cea-144f-445c-b3b6-c16b65d4c274.jpg)
Rotate device to portrait mode and choose actions again - now there are update and download buttons:
![image007](https://user-images.githubusercontent.com/39751382/135737252-39a87a81-8f6d-4e91-95e2-497f1ae03dea.jpg)
Rotate back to landscape and choose actions again - now we have update and download buttons in landscape mode:
![image008](https://user-images.githubusercontent.com/39751382/135737268-e564e0e6-a520-4c87-a00d-cf56c12c52a9.jpg)

In either orientation the download button just gives a toast saying "This map could not be downloaded". The update button produces a confirmation dialog "Update local data from Internet", and if I choose "UPDATE MAP" instead of "CANCEL" then I get the same toast.

From the main map download menu I can get the standard map for this region (France Ile-de-France) without problems:
![image001](https://user-images.githubusercontent.com/39751382/135737349-5e7f0ec8-398d-48a8-8158-73ca302ceef2.jpg)

My device: Galaxy Tab A SM-T510, Android10.

Thanks very much for further assistance, or for any actions the devs may take.

**Reply by tommycrock** [reply]
I'm case it helps, my device is also a Samsung Galaxy, but a s10e with Android 11. I don't get those quick action buttons even in portrait mode:
![Screenshot_20211003-065933_OsmAnd+](https://user-images.githubusercontent.com/5749884/135742058-ce52a30c-d0e4-4177-963e-c8c563a86dad.jpg)
Thanks

**Comment by thompehw** [comment, marked as answer]
Given lack of further interest from devs, I wondered if it was fixed in the latest nightly build. It's different, and functions after a long-press (no need to pick "actions"). Looks like this is under active development, so I will flag this questions as answered.
